src/sql/src/plan/statement/ddl.rs

```diff
@@ -387,7 +389,16 @@ pub fn plan_create_source(
                 Some(v) => bail!("invalid start_offset value: {}", v),
             }
 
-            let encoding = get_encoding(scx, format, &envelope, with_options_original)?;
+            let encoding = get_encoding(
+                scx,
+                format,
+                &envelope,
+                with_options_original,
+                secrets_reader,
+            )?;
+
+            // XXX(chae): is this the place to inline? Or should it get delayed until we actually connect to kafka?
```

Comment:
Thoughts? I haven't thought about it deeply but I believe delaying inlining would require passing through a copy of the `SecretsReader` with `KafkaSourceConnector`

Reply:
hmm I'd like to get away from the model of changing the types within a struct.  If we don't pass through the `SecretsReader`, I'd imagine we'd do one of the following:
- Fill in a `Option<BTreeMap<String,String>>` (this seems bad and is something nikhil got rid of recently)
- Change all the `MaybeStringId::Secret`s to be `Strings` and assert later that everything is the String variant

If we pass the `MaybeStringId`s along further, I'd prefer we do it all the way through with a `SecretsReader` -- now that we've made `SecretsReader` impl `Clone`!

What do you think?

Reply:
Ah, I think it could be done on `CatalogState` itself, which already has the `SecretsReader`, i.e. `CatalogState::source_description_for`. That is basically as late as humanly possible. The question remains, though, what the types ought to be. I'm not opposed to literalizing them (i.e. taking them all to `String`), but idk if that's too hacky.

Reply:
I did a little digging here and doing processing at that point would require a lot of manipulation to do the translation in that function / somewhere around there because the mapping is so many layers deep inside of `CatalogItem`.

And then doing the inline even later, in `render_source`, e.g., would require moving the secrets reader into storage also -- which we probably don't want to do.

From our chat earlier, it seems like you're okay with this so I'll leave it as is

Reply:
> And then doing the inline even later, in render_source, e.g., would require moving the secrets reader into storage also -- which we probably don't want to do.

No no, this is _exactly_ what we want to do! We should avoid sending secrets over the wire whenever possible, and the tighter the scope is on a secret that comes out of the reader, the better. Pushing the secret reading into storaged is a great way to accomplish this.

Totally fine to leave this to a follow up PR though.

Reply:
sure -- will leave a TODO and an issue